internal/hcp/registry/json.go

```diff
@@ -93,7 +100,13 @@ func (h *JSONRegistry) CompleteBuild(
 	artifacts []sdkpacker.Artifact,
 	buildErr error,
 ) ([]sdkpacker.Artifact, error) {
-	return h.bucket.completeBuild(ctx, build.Name(), artifacts, buildErr)
+	buildName := build.Name()
+	buildMetadata := build.(*packer.CoreBuild).GetMetadata()
+	err := h.bucket.Version.AddMetadataToBuild(ctx, buildName, buildMetadata)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I wonder if we could pass the build instance to the methods we call here, this piece of code is quadrupled in the code (twice for JSON for start/complete, same for HCL), this looks factorisable, and I believe we can migrate this logic to the registry package.

Reply:
Nevermind this then, after out-of-band discussing it appears that because of how we switch names between legacy JSON/HCL2 it cannot be trivially factorised, so we can keep it this way for a bit, but we should fix that at some point.